ykrt/src/trace/hwt/mapper.rs

```diff
@@ -147,7 +147,10 @@ impl<'a> HWTMapper {
     ) -> Result<Vec<IRBlock>, HWTracerError> {
         let mut ret: Vec<IRBlock> = Vec::new();
 
-        for block in &mut trace_iter {
+        for (i, block) in &mut trace_iter.enumerate() {
+            if i > crate::mt::DEFAULT_TRACE_TOO_LONG {
```

Comment:
Hrm. I looked to see if we can avoid doing *any* mapping at all if the trace is too long, but the decoder is lazy and we don't really know without decoding the trace...

You could express the acceptable trace length in "bytes of PT payload", but that wouldn't be portable between trace collectors...

i guess you had similar thoughts?

Reply:
I did. Clearly we'll need to do something better than this at some point (e.g. we could do an analysis of PT buffer lengths and their average number of blocks, and guess based on that). But that's a lot of complexity and if we optimise some of the parts of, for example, the mapper, we'd have to reanalyse everything again. So I thought: a simple, obviously crude, threshold is probably a tolerable starting point. It has the virtue of being easy to adjust!